Client/Application/UserConversionMetrics.swift

```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Common
+import Foundation
+import Shared
+import StoreKit
+
+class UserConversionMetrics {
+    private var appOpenTimestamps: [Date] {
+        get { return userDefaults.array(forKey: PrefsKeys.Session.firstWeekAppOpenTimestamps) as? [Date] ?? [Date]() }
+        set { userDefaults.set(newValue, forKey: PrefsKeys.Session.firstWeekAppOpenTimestamps) }
+    }
+
+    private var searchesTimestamps: [Date] {
+        get { return userDefaults.array(forKey: PrefsKeys.Session.firstWeekSearchesTimestamps) as? [Date] ?? [Date]() }
+        set { userDefaults.set(newValue, forKey: PrefsKeys.Session.firstWeekSearchesTimestamps) }
+    }
+    var userDefaults: UserDefaultsInterface = UserDefaults.standard
+
+    init(userDefaults: UserDefaults = .standard) {
+        self.userDefaults = userDefaults
+    }
+
+    public func didStartNewSession() {
+        if shouldRecordMetric() {
+            appOpenTimestamps.append(Date())
+            if shouldActivateProfile() {
+                sendActivationEvent()
+            }
+        }
+    }
+
+    public func didPerformSearch() {
+        if self.shouldRecordMetric() {
+            searchesTimestamps.append(Date())
+            if shouldActivateProfile() {
+                sendActivationEvent()
+            }
+        }
+    }
+
+    // there shouldn't be more records added to UserDefaults once this event happens and it should not affect existing users
```

Comment:
For documentation we use ///
```suggestion
    /// There shouldn't be more records added to UserDefaults once this event happens and it should not affect existing users
```